src/Data/String/CodePoints.purs

```diff
@@ -0,0 +1,225 @@
+-- | These functions allow PureScript strings to be treated as if they were
+-- | sequences of Unicode code points instead of their true underlying
+-- | implementation (sequences of UTF-16 code units). For nearly all uses of
+-- | strings, these functions should be preferred over the ones in Data.String.
+module Data.String.CodePoints
+  ( module StringReExports
+  , CodePoint()
+  , codePointAt
+  , codePointFromInt
+  , codePointToInt
+  , count
+  , drop
+  , dropWhile
+  , fromCodePointArray
+  , indexOf
+  , indexOf'
+  , lastIndexOf
+  , lastIndexOf'
+  , length
+  , singleton
+  , splitAt
+  , take
+  , takeWhile
+  , toCodePointArray
+  , uncons
+  ) where
+
+import Prelude
+
+import Data.Array as Array
+import Data.Char as Char
+import Data.List (List(Cons, Nil), fromFoldable)
+import Data.Maybe (Maybe(Just, Nothing))
+import Data.Newtype (class Newtype)
+import Data.String as String
+import Data.String.Unsafe as Unsafe
+-- WARN: This list must be updated to re-export any exports added to Data.String. That makes me sad.
```

Comment:
The idea is that this module exports a superset of the things exported by `Data.String` overall, so that it's a drop-in replacement, correct? And the functions in this list are just the functions exported by `Data.String` which don't care whether a string is treated as a sequence of code points or code units? Because in that case the comment we have here currently doesn't seem entirely correct.